### PR TITLE
Wlu/optimization calculations

### PR DESCRIPTION
**Pull request by wirooo**
No longer throw out values that exceed max race time. Now use profiles that are too long to work towards optimal solution. Also made sure to check final solution does not exceed max time before printing out.

### CODE REVIEW

File: optimization/generate_profile.py [hunk 1]
```diff
@@ -105,38 +102,51 @@ def load_course_map(course_name="COTA"):
...
     return elev_profile
  
 if __name__ == '__main__':
     elev_profile = load_course_map()
     # Load in the distance and necessary time for a lap
     distance = 5490
-    time = 420 # max allowable time in s
-    iterations = 500  # Number of iterations to use
+    time = 420
+    iterations = 500 # Number of iterations to use
```
Comment (wirooo) [inline optimization/generate_profile.py]: need to change iterations to whatever num it should be

File: optimization/generate_profile.py [hunk 2]
```diff
@@ -51,14 +51,11 @@ def calculate_fit(v_profile, e_profile, max_time=450, dist_step=30):
...
-    if sum(time) > max_time:
-        fit = -1 # TODO: allow smart usage of too slow runs
-    else: 
-        fit = energy
+    fit = energy
     return [fit, sum(time)]
 
 
-def generate_new_profile(v_profile, e_profile, dist_step=30, min_v = 5):
+def generate_new_profile(v_profile, e_profile, dist_step=30, min_v=4):
```
Comment (wirooo) [inline optimization/generate_profile.py]: need to change back min_v

File: optimization/generate_profile.py (hunk 1 above)
Comment (clarkevandenhoven) [reply optimization/generate_profile.py]: I'd leave in the time comment. As well, maybe make that a passed-in parameter

File: optimization/generate_profile.py (hunk 2 above)
Comment (clarkevandenhoven) [reply optimization/generate_profile.py]: Min_V should also be a parameter passed by the user

File: optimization/generate_profile.py [hunk 3]
```diff
@@ -105,38 +102,51 @@ def load_course_map(course_name="COTA"):
...
+    iterations = 500 # Number of iterations to use
     init_profile = generate_initial_profile(time, distance, elev_profile)
     v_profile = init_profile
+    last_real_solution = v_profile
     values = []
     # Calculate the fitness of each population:
     values.append(calculate_fit(init_profile, elev_profile))
     min_value = 1000000
     min_time = 1000000
+    max_race_time = 1000
```
Comment (clarkevandenhoven) [inline optimization/generate_profile.py]: This should be user-given

File: optimization/generate_profile.py (hunk 1 above)
Comment (wirooo) [reply optimization/generate_profile.py]: Are time and max_race_time supposed to be different?

File: optimization/generate_profile.py (hunk 3 above)
Comment (wirooo) [reply optimization/generate_profile.py]: How should we implement user-given values? I imagine its not just input(). Would we want to read from a separate conifg file? Not sure what the standard is.

File: optimization/generate_profile.py (hunk 1 above)
Comment (clarkevandenhoven) [reply optimization/generate_profile.py]: Note here. They shouldn't be. Appears to be an artifact of testing. Should be fixed.

**Pull request merged**